pkgs/servers/typedb/default.nix

```diff
@@ -0,0 +1,97 @@
+{ stdenv, lib, openjdk, system, fetchzip, fetchurl, server-data ? "~/.typedb_home", server-port ? "1729", server-logs ? "~/.typedb_home/logs" }:
+let
+  typedbVersion = "2.1.1";
+  systems = {
+    x86_64-linux = {
+      src = fetchurl {
+        url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-linux-2.1.1.tar.gz";
+        sha256 = "00wvkiahmrys32ckp6r59bbrzfd7jzsphniyjdj4p9bh7k339gn2";
+      };
+      dir = "typedb-all-linux-${typedbVersion}";
+    };
+    x86_64-darwin = {
+      src = fetchzip {
+        url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-mac-2.1.1.zip";
+        sha256 = "16hlfy6kh2rnvcralz206q13mghb0rv8wazpg6q3h324p5rdys54";
+      };
+      dir = "typedb-all-mac-${typedbVersion}";
+    };
+    x86_64-windows = {
+      src = fetchzip {
+        url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-windows-2.1.1.zip";
+        sha256 = "0vd66gfshkg697z07nhy957mwqzlli4r4pmn67hx58n9mkg024kq";
+      };
+      dir = "typedb-all-windows-${typedbVersion}";
+    };
+  };
+
+  currentSystem = systems.${system} or throw "unsupported system ${system}";
+  typedbDir = currentSystem.dir;
+  srcFolder = currentSystem.src;
+  javaPatch = ''
+    20c20
+    < JAVA_BIN=java
+    ---
+    > JAVA_BIN=${openjdk}/bin/java
+  '';
+  typedb-properties-patch = ''
+    19c19
+    < server.data=server/data/
+    ---
+    > server.data=${server-data}
+    22c22
+    < server.logs=server/logs/
+    ---
+    > server.logs=${server-logs}
+    24c24
+    < server.port=1729
+    ---
+    > server.port=${server-port}
```

Comment:
While it's already much better, this is still a bit awkward. Ideally changes in configuration would not require a rebuild.
If we're lucky, we might still be able to use an environment variable instead of hardcoding it into the store path, although that's not the default property file behavior.

Somewhat relevant discussion https://stackoverflow.com/questions/2263929/regarding-application-properties-file-and-environment-variable. If it's built on apache commons, we're lucky.

Reply:
Is it possible to specify properties or property files at runtime?

Reply:
according to the typedb team not;
only at startup as a cli option or in the config file;

I decided to add a wrapper script again that makes it possible to change the three configuration options via environment variables.
at least until I am able to change the implementation of the property file reader in the typedb.

I am still debugging it, but at least it builds and the tests run fine now;
there are still some issues though...

Reply:
> only at startup as a cli option or in the config file;

The config file is not really a config file if it's packaged immutably with the software, so cli options are probably the easiest option. Do you plan to write a NixOS service as well?